contracts/programs/gummyshop/src/lib.rs

```diff
@@ -0,0 +1,356 @@
+use anchor_lang::{
+    prelude::*,
+    solana_program::{
+        keccak::hashv, 
+        sysvar,
+        sysvar::instructions::{load_instruction_at_checked}, 
+        sysvar::SysvarId, 
+        pubkey::Pubkey, 
+        program::{invoke, invoke_signed},
+        system_instruction,
+        instruction::{Instruction},
+    },
+};
+use anchor_spl::token::{Mint, TokenAccount, Token, Transfer, transfer};
+use bubblegum::program::Bubblegum;
+use gummyroll::program::Gummyroll;
+use bubblegum::state::{
+    leaf_schema::Version,
+    metaplex_adapter::{MetadataArgs},
+};
+
+declare_id!("9T5Xv2cJRydUBqvdK7rLGuNGqhkA8sU8Yq1rGN7hExNK");
+
+// TODO(sorend): port to a state directory/mod
+#[account]
+#[derive(Copy)]
+pub struct MarketplaceProperties {
+    // The address to which the marketplace will receive royalties
+    pub payment_receiver: Pubkey,
+    // The royalty percentage (not basis points) the marketplace will receive upon purchase of an NFT
```

Comment:
Let's make this bps, 1+% marketplace fees already disgust me

Reply:
LOL sounds good